### PR TITLE
271 4.14 additional application field

### PR DESCRIPTION
**Pull request by phnm**
#271

### CODE REVIEW

File: app/controllers/emails_controller.rb
Comment (bjrne) [inline app/controllers/emails_controller.rb:10]: How is this change related to the topic of this PR?

File: db/migrate/20170124225558_add_annotation_to_application_letters.rb
Comment (bjrne) [inline db/migrate/20170124225558_add_annotation_to_application_letters.rb]: Please use text here, this could end in a lot more than 255 characters

File: spec/controllers/events_controller_spec.rb
Comment (bjrne) [inline spec/controllers/events_controller_spec.rb:227]: Please don't commit so many whitespace changes

File: spec/features/participants_spec.rb
Comment (bjrne) [inline spec/features/participants_spec.rb]: same here

File: spec/models/event_spec.rb
Comment (bjrne) [inline spec/models/event_spec.rb:95]: And why is this change in this PR. I don't see it related to this in any way

File: spec/models/event_spec.rb
Comment (bjrne) [inline spec/models/event_spec.rb:194]: again. Why in this PR? Please open separate PRs for separate topics.

**Review by bjrne — CHANGES_REQUESTED** [review]
I would approve this, if only features related to your issue were changed. However I don't like all the whitespace changes and things that have nothing to do with this. Or is this just a mistake and those changes came from dev?

File: spec/models/event_spec.rb
Comment (phnm) [reply spec/models/event_spec.rb:95]: You're right. But Travis does. This is a long story, but for some reason 3 tests were always failing only on travis and always passing on local. Even though I didn't touch the test or the feature they test, as you can see in the commit history of my branch. Don't ask me why, but I had to refactor those tests in order to have them passing on Travis.

File: spec/controllers/events_controller_spec.rb
Comment (phnm) [reply spec/controllers/events_controller_spec.rb:227]: My editor (Atom) automatically deletes trailing whitespaces in files I touch. Is that a bad thing?

File: db/migrate/20170124225558_add_annotation_to_application_letters.rb
Comment (phnm) [reply db/migrate/20170124225558_add_annotation_to_application_letters.rb]: Good point, I will change this.

File: app/controllers/emails_controller.rb
Comment (phnm) [reply app/controllers/emails_controller.rb:10]: I had to refactor the email_addresses_of_type(type) method, since it was returning a string with concatenated emails addresses in a random order on Travis. Only Travis knows why this wasn't an issue earlier, but tests were suddenly faling because of this. I changed it to return an array of Emails, which is better anyway, but made this change to join the addresses here necessary.

File: spec/models/event_spec.rb
Comment (bjrne) [reply spec/models/event_spec.rb:95]: I'd have liked to see a separate small PR for this, followed by the feature one. This way, the 'travis'-PR would have maybe attracted some other people for reviewing and commenting (and noticing) than hiding it in this PR did. Things like this may be of interest for more developer. Do you know now exactly why this change solves the problem?

File: spec/controllers/events_controller_spec.rb
Comment (bjrne) [reply spec/controllers/events_controller_spec.rb:227]: not necessarily, but I like to only have the productive changes in the feature PRs in order to be able to focus on those. If people constantly leave trailing whitespace and editors change it forth and back it might be a good idea to have a cleanup PR where you fix a lot of those formatting problems.

File: app/controllers/emails_controller.rb
Comment (bjrne) [reply app/controllers/emails_controller.rb:10]: Hm, this is more refactoring than I'd like in a PR for another topic. I'd like some feedback from somebody else, maybe from your team or from the email people @frederike-ramin @aloeser @Baschdl. I would really like this to be separated from this PR. It shouldn't be too much effort with `git stash`.

File: spec/models/event_spec.rb
Comment (phnm) [reply spec/models/event_spec.rb:95]: I know what went wrong, but have no idea why now and why only on travis. I also posted this in issue #374 since this might be related, but nobody seemed to have an idea.
The problem with this test was that, when comparing a string with `eq` there is an assumed order of the expected concatenated email addresses. But those addresses were never sorted, they just happend to be in the expected order so far. And that order was different on Travis on my branch.

Comment (bjrne) [reply spec/models/event_spec.rb:95]: Hm, okay. Would you be willing to open a separate PR for this, link it to the issue and once it is merged, we merge this one? I feel like these are two full features, since your refactoring also changed a little bit of logic.

Comment (phnm) [reply spec/models/event_spec.rb:95]: Sure, see #441

**Review by bjrne — APPROVED** [review]
Looks good now 🏎

**Pull request merged**
